### ISSUE TITLE
Mark Event as Failed (do not retry)

### ISSUE THREAD

**Issue by ticolucci** [klarna/phobos_db_checkpoint#7]
Sometimes we have exceptions that won't be fixed and we shouldn't retry (blocking the partition forever).

So @tulios and I propose to have a return message from the `Handler#consume`, similar to the `ack`, called `reject` (unfortunately `fail` is a Ruby  core method...).

Example:

``` ruby
class LeHandler
  include PhobosDBCheckpoint::Handler

  def consume(payload, metadata)
    do_something_cool
  rescue UnprocessableEventError => exception
    reject(exception)
  end
end
```

The idea is to create a separate table called `Failures` where we store the `payload`, `metadata`, `handler`, and the `exception` for further debugging.

One should be able to retry a `Failure` (a simple ActiveRecord model) by simply calling `Failure#retry`.

**Comment by tulios** [comment]
We could expose failures with #9

**Comment by klippx** [comment]
It would indeed be cool to have Failure handling in this repo, so it doesn't have to be implemented ad hoc by the Phobos apps. It would also make it easier to handle retries and failures in the Events API, and displaying failures in the graphical interface (OST v2) for any Phobos DB Checkpoint user.

**Comment by klippx** [comment]
Fixed in #13
